src/Testing/CoreTests/ErrorHandling/FailureRuleCollectionTests.cs

```diff
@@ -259,6 +259,32 @@ public void schedule_retry_n_times()
         theHandlers.Failures.DetermineExecutionContinuation(new BadImageFormatException(), theEnvelope)
             .ShouldBeOfType<MoveToErrorQueue>();
     }
+    
+    [Fact]
+    public void schedule_indefinite_retries()
+    {
+        theHandlers.OnException<BadImageFormatException>().ScheduleIndefiniteRetry(1.Minutes(), 5.Minutes(), 10.Minutes());
+
+        theEnvelope.Attempts = 0;
+        theHandlers.Failures.DetermineExecutionContinuation(new BadImageFormatException(), theEnvelope)
+            .ShouldBeOfType<ScheduledRetryContinuation>().Delay.ShouldBe(1.Minutes());
+
+        theEnvelope.Attempts = 1;
+        theHandlers.Failures.DetermineExecutionContinuation(new BadImageFormatException(), theEnvelope)
+            .ShouldBeOfType<ScheduledRetryContinuation>().Delay.ShouldBe(1.Minutes());
+
+        theEnvelope.Attempts = 2;
+        theHandlers.Failures.DetermineExecutionContinuation(new BadImageFormatException(), theEnvelope)
+            .ShouldBeOfType<ScheduledRetryContinuation>().Delay.ShouldBe(5.Minutes());
+
+        theEnvelope.Attempts = 3;
+        theHandlers.Failures.DetermineExecutionContinuation(new BadImageFormatException(), theEnvelope)
+            .ShouldBeOfType<IndefiniteScheduledRetryContinuation>().Delay.ShouldBe(10.Minutes());
+
+        theEnvelope.Attempts = 4;
+        theHandlers.Failures.DetermineExecutionContinuation(new BadImageFormatException(), theEnvelope)
+            .ShouldBeOfType<MoveToErrorQueue>();
```

Comment:
Just to comment, this section here is a bit confusing because the failure slots list will be expanded when the IndefiniteSchedluledRetryContinuation is executed, thus the scenario when `attempts == 4` could be a bit misleading in terms of the text, maybe :)